### PR TITLE
Handle terms and pp events on confirm page (bug 1133587)

### PR DESCRIPTION
**Pull request by muffinresearch**

### CODE REVIEW

**Comment by kumar303** [comment]
might be a legitimate test failure

File: tests/ui/test-fxa-iframed-pp.js [hunk 1]
```diff
@@ -52,6 +52,21 @@ casper.test.begin('Check iframed FxA privacy link', {
 
     casper.waitWhileSelector('.iframe-overlay', function() {
       test.assertNotVisible('.iframe-overlay iframe', 'Check overlay is removed');
+      this.click('.pinbox');
+      this.sendKeys('.pinbox', '1234');
+      test.assertExists('.cta:enabled', 'Submit button is enabled');
+      this.click('.cta');
+      casper.click('.pp');
```
Comment (kumar303) [inline tests/ui/test-fxa-iframed-pp.js]: not ``this.click()`` ? Maybe a comment explaining why?

File: tests/ui/test-fxa-iframed-pp.js [hunk 2]
```diff
@@ -52,6 +52,21 @@ casper.test.begin('Check iframed FxA privacy link', {
...
+    });
+
+    casper.waitForSelector('.iframe-overlay', function() {
+      // Check that the link works on the confirm pin page.
+      casper.echo(casper.getElementAttribute('.iframe-overlay iframe', 'src'));
```
Comment (kumar303) [inline tests/ui/test-fxa-iframed-pp.js]: debugging? If not, maybe prefix it to describe what it is

File: tests/ui/test-fxa-iframed-pp.js [hunk 3]
```diff
@@ -52,6 +52,21 @@ casper.test.begin('Check iframed FxA privacy link', {
...
+      test.assertMatch(
+        casper.getElementAttribute('.iframe-overlay iframe', 'src'),
+        /privacy\/en-US\.html/,
```
Comment (kumar303) [inline tests/ui/test-fxa-iframed-pp.js]: I assume this iframe loaded URL on the test server? The tests should not load external URLs.

File: tests/ui/test-fxa-iframed-terms.js [hunk 4]
```diff
@@ -52,6 +52,21 @@ casper.test.begin('Check iframed FxA terms link', {
...
+      this.click('.pinbox');
+      this.sendKeys('.pinbox', '1234');
+      test.assertExists('.cta:enabled', 'Submit button is enabled');
+      this.click('.cta');
+      casper.click('.tos');
+    });
+
+    casper.waitForSelector('.iframe-overlay', function() {
+      // Check that the link works on the confirm pin page.
+      casper.echo(casper.getElementAttribute('.iframe-overlay iframe', 'src'));
```
Comment (kumar303) [inline tests/ui/test-fxa-iframed-terms.js]: also maybe looks like debugging that can be removed

**Comment by kumar303** [comment]
r+wc once the tests pass

File: tests/ui/test-fxa-iframed-pp.js (hunk 1 above)
Comment (muffinresearch) [inline tests/ui/test-fxa-iframed-pp.js]: `this` is `casper` here, but yeah could be `this` for consistency.

File: tests/ui/test-fxa-iframed-pp.js (hunk 3 above)
Comment (muffinresearch) [inline tests/ui/test-fxa-iframed-pp.js]: yeah, I noticed that before - this is loading a real url. I'll fix that up as part of this patch next week.

**Comment by muffinresearch** [comment]
> might be a legitimate test failure

This is something timing related. I've seen this test die before intermittently on slimer.

**Comment by kumar303** [comment]
aw, bummer. Maybe we can put something in a wait-for-something loop to make it more robust.

**Comment by muffinresearch** [comment]
@kumar303 this now doesn't call the external terms URLS any more.

File: config/test.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ module.exports = {
   useMinCSS: true,
   useMinJS: true,
   showClientConsole: false,
-  staticURL: 'https://marketplace-dev-cdn.allizom.org/',
+  staticURL: 'http://localhost:2599/',
```
Comment (kumar303) [inline config/test.js]: :+1: :tada:

**Comment by kumar303** [comment]
r+

File: config/test.js (hunk 5 above)
Comment (kumar303) [inline config/test.js]: is this URL and port stable enough to hard code here?

Comment (muffinresearch) [inline config/test.js]: yeah - it's specific to testing.

**Pull request merged**